paddle/fluid/operators/mkldnn/fc_mkldnn_op.cc

```diff
@@ -405,22 +147,20 @@ class FCPrimitiveFactory {
   // those scales to normalise them back to what their floating-point range
   // was. Then we multiply them by desired output scale we want on the output.
   std::vector<float> ComputeOutputShiftScale(const ExecutionContext& ctx) {
-    auto scale_in_data = ctx.Attr<float>("Scale_in");
-    auto scale_weights_data = ctx.Attr<std::vector<float>>("Scale_weights");
+    auto scale_in = ctx.Attr<float>("Scale_in");
+    auto scale_weights = ctx.Attr<std::vector<float>>("Scale_weights");
     // If the output will be in floats, we don't multiply by scale_out.
-    auto scale_out_data = ctx.Attr<bool>("force_fp32_output")
-                              ? 1.0f
-                              : ctx.Attr<float>("Scale_out");
-    const size_t weight_scales_num = scale_weights_data.size();
-    std::vector<float> output_shift_scale(weight_scales_num);
+    auto scale_out = ctx.Attr<bool>("force_fp32_output")
+                         ? 1.0f
+                         : ctx.Attr<float>("Scale_out");
+    std::vector<float> output_shift_scale(scale_weights.size());
 
 #pragma omp parallel for
-    for (size_t i = 0; i < weight_scales_num; i++) {
-      if (scale_weights_data[i] == 0.0)
-        output_shift_scale[i] = scale_out_data;
+    for (size_t i = 0; i < output_shift_scale.size(); ++i) {
```

Comment:
Please add caching of scales of weights and remove omp parallel after caching is done